### PR TITLE
Design refresh: Update PageHeader and move to Wildcard

### PR DESCRIPTION
**Pull request by umpox**
This PR:
- Updates the `PageHeader` layout and styling to match the design refresh
- Removes the dependency on Bootstrap, and migrates styling to CSS modules
- Implements the updated `description` field on a few different pages.

image

closes https://github.com/sourcegraph/sourcegraph/issues/19932

### CODE REVIEW

File: client/web/src/enterprise/codeintel/list/CodeIntelIndexesPage.tsx [hunk 1]
```diff
@@ -80,7 +80,7 @@ export const CodeIntelIndexesPage: FunctionComponent<CodeIntelIndexesPageProps>
             <PageTitle title="Precise code intelligence auto-index records" />
             <PageHeader
                 path={[{ text: 'Precise code intelligence auto-index records' }]}
-                byline={
+                description={
```
Comment (umpox) [inline client/web/src/enterprise/codeintel/list/CodeIntelIndexesPage.tsx:84]: We don't have Percy snapshots here yet:

| BEFORE             |  AFTER |
:-------------------------:|:-------------------------:
image | image

File: client/web/src/enterprise/codeintel/list/CodeIntelUploadsPage.tsx [hunk 2]
```diff
@@ -107,7 +107,7 @@ export const CodeIntelUploadsPage: FunctionComponent<CodeIntelUploadsPageProps>
             <PageTitle title="Precise code intelligence uploads" />
             <PageHeader
                 path={[{ text: 'Precise code intelligence upload' }]}
-                byline={
+                description={
```
Comment (umpox) [inline client/web/src/enterprise/codeintel/list/CodeIntelUploadsPage.tsx:110]: We don't have Percy snapshots here yet:

| BEFORE             |  AFTER |
:-------------------------:|:-------------------------:
image | image

File: client/wildcard/src/components/PageHeader/PageHeader.module.scss [hunk 3]
```diff
@@ -0,0 +1,70 @@
+// Scale icon size with heading size. 26px font size (h1) -> 22px icon size
+$path-icon-size: (22em / 26);
+
+.container {
+    display: flex;
+    flex-direction: column;
+    flex-wrap: wrap;
+    justify-content: space-between;
+
+    @media (min-width: 768px) {
```
Comment (umpox) [inline client/wildcard/src/components/PageHeader/PageHeader.module.scss:10]: @sourcegraph/frontend-platform Thoughts on how we should we handle these kind of constants in Wildcard?

It'd be nice to not have to introduce Bootstrap here (and ensures we can still easily use this code across both internal and external environments). Maybe we should define a `variables.scss` file in Wildcard that is effectively just SCSS vars that are replaced once compiled?

Comment (eseliger) [reply client/wildcard/src/components/PageHeader/PageHeader.module.scss:10]: Can we have our own mixin for that? Currently we import it from bootstrap elsewhere, but maybe we can extract the code for them into the wildcard package.

**Review by eseliger — APPROVED** [review]
Nice 👍

File: client/wildcard/src/components/PageHeader/PageHeader.module.scss (hunk 3 above)
Comment (valerybugakov) [reply client/wildcard/src/components/PageHeader/PageHeader.module.scss:10]: Continuing the [thread](https://github.com/sourcegraph/sourcegraph/pull/20620#discussion_r626335814) about leaning more toward CSS variables and CSS spec, maybe we can consider something like: https://github.com/postcss/postcss-custom-media for media queries in the Wildcard package?

It follows the CSS Media Queries specification:

```css
@custom-media --small-viewport (max-width: 30em);

@media (--small-viewport) {
  /* styles for small viewport */
}

/* becomes */

@media (max-width: 30em) {
  /* styles for small viewport */
}

**Review by valerybugakov — APPROVED** [review]
The new addition to the Wildcard family 🎉

**Review by rrhyne — APPROVED** [review]
I couldn't find any issues from the design side!

File: client/wildcard/src/components/PageHeader/PageHeader.module.scss (hunk 3 above)
Comment (umpox) [reply client/wildcard/src/components/PageHeader/PageHeader.module.scss:10]: Can't go wrong by following the spec more, going to raise another PR with support for this!

**Pull request merged**